src/Debugger/Debuggee.php

```diff
@@ -0,0 +1,285 @@
+     *            version ID is structured as following:
+     *            domain/type/vmajor.minor (for example
+     *            google.com/gcp-java/v1.1).
+     *      @type string $status Human readable message to be displayed to the
+     *            user about this debuggee. Absence of this field indicates no
+     *            status. The message can be either informational or an error
+     *            status.
+     *      @type ExtendedSourceContext[] $extSourceContexts References to the locations and
+     *            revisions of the source code used in the deployed application.
+     */
+    public function __construct(ConnectionInterface $connection, array $info = [])
+    {
+        $this->connection = $connection;
+
+        $info += [
+            'id' => null,
+            'isInactive' => false,
+            'agentVersion' => DebuggerClient::DEFAULT_AGENT_VERSION,
+            'status' => null,
+            'extSourceContexts' => [],
+            'uniquifier' => null,
+            'description' => null
+        ];
+
+        $this->id = $info['id'];
+        $this->project = $info['project'];
+        $this->uniquifier = $info['uniquifier'];
+        $this->description = $info['description'];
+        $this->status = $info['status'];
+        $this->agentVersion = $info['agentVersion'];
+        $this->isInactive = $info['isInactive'];
+        $this->extSourceContexts = $info['extSourceContexts'];
+    }
+
+    /**
+     * Return the debuggee identifier.
+     *
+     * Example:
+     * ```
+     * echo $debuggee->id();
+     * ```
+     *
+     * @return string
+     */
+    public function id()
+    {
+        return $this->id;
+    }
+
+    /**
+     * Register this debuggee with the Stackdriver backend.
+     *
+     * Example:
+     * ```
+     * $debuggee->register();
+     * echo $debuggee->id();
+     * ```
+     *
+     * @param array $args
+     * @return bool
+     */
+    public function register(array $args = [])
+    {
+        $resp = $this->connection->registerDebuggee(['debuggee' => $this] + $args);
+        if (array_key_exists('debuggee', $resp)) {
+            $this->id = $resp['debuggee']['id'];
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Fetch the list of breakpoints this debugee should try to handle.
+     *
+     * Example:
+     * ```
+     * $breakpoints = $debuggee->breakpoints();
+     * ```
+     *
+     * @param array $options [optional] {
+     *      Configuration options.
+     *
+     *      @type string $waitToken A wait token that, if specified, blocks the
+     *            method call until the list of active breakpoints has changed,
+     *            or a server selected timeout has expired. The value should be
+     *            set from the last returned response.
+     *      @type bool $successOnTimeout If set to **true**, returns
+     *            **google.rpc.Code.OK** status and sets the **waitExpired**
```

Comment:
These codes could end up not meaning much for the REST transport. Should we reword this a bit?

Reply:
I'll remove this option - this case isn't actually handled.